editor/src/DocumentWindows/MainScene.cpp

```
@@ -33,11 +34,12 @@
 namespace nexo::editor {
 
-    MainScene::MainScene(std::string sceneName, const bool defaultScene) : m_sceneName(std::move(sceneName)),
-                                                                           m_defaultScene(defaultScene) {}
-
-    MainScene::~MainScene()
-    = default;
+    MainScene::MainScene(WindowRegistry &windowRegistry, std::string sceneName, const bool defaultScene) : m_sceneName(std::move(sceneName)),
+                                                                           m_defaultScene(defaultScene),
+                                                                           m_windowRegistry(windowRegistry)
+    {
+    	windowId = nextWindowId++;
```

Cppcheck (2.10-2)

[performance] Variable 'windowId' is assigned in constructor body. Consider performing initialization in initialization list.

(useInitializationList)